herald.py

```diff
@@ -22,19 +27,34 @@ async def runHerald(ctx, args):
         except:
             await ctx.send("Not a valid YouTube link")
             return
+
+        # Video cannot be longer than 10 minutes
+        if video.length > 600:
+            await ctx.send("Video is too long (greater than 10 minutes)")
+            return
         
         stream = video.streams.filter(only_audio = True).first()
-        stream.download("herald", filename = f'{ctx.author.id}_audio.mp3')
+        stream.download("herald", filename = f'{ctx.author.id}.mp3')
 
         if ctx.author.id in heraldDict.keys():
             # User exists
-            heraldDict[ctx.author.id].mp3Link = f'herald/{ctx.author.id}_audio.mp3'
+            heraldDict[ctx.author.id].mp3Link = f'herald/{ctx.author.id}.mp3'
             await ctx.send("Herald updated")
         else:
             # New user
-            heraldDict[ctx.author.id] = HeraldUser(f'herald/{ctx.author.id}_audio.mp3')
+            heraldDict[ctx.author.id] = HeraldUser(f'herald/{ctx.author.id}.mp3')
             await ctx.send("User initialized with new Herald")
 
+        # Ensure the playback does not last longer than possible
+        if video.length < default_duration:
+            heraldDict[ctx.author.id].duration = video.length
+        else:
```

Comment:
else block not technically needed since HeraldUser object is initialized with default_duration already.

Reply:
I put this here in the case of an existing user choosing a new audio, it would reset it to the default duration.

Reply:
Ah, good thinking.